pyth-sdk/src/price.rs

```diff
@@ -897,4 +1205,801 @@ mod test {
+            1000,
+            900,
+            -3,
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9)
+        );
+
+        // half deposits, diff precision
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            50,
+            100,
+            1000,
+            900,
+            -3,
+            pc(95 * (PD_SCALE as i64), 2 * PD_SCALE, -9)
+        );
+
+        // full deposits, diff precision
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            100,
+            100,
+            1000,
+            900,
+            -3,
+            pc(90 * (PD_SCALE as i64), 2 * PD_SCALE, -9)
+        );
+
+        // beyond final endpoint deposits
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            150,
+            100,
+            100,
+            90,
+            -2,
+            pc(85 * (PD_SCALE as i64), 2 * PD_SCALE, -9)
+        );
+
+        // 0 deposits, staggered initial discount
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            0,
+            100,
+            98,
+            90,
+            -2,
+            pc(98 * (PD_SCALE as i64), 2 * PD_SCALE, -9)
+        );
+
+        // half deposits, staggered initial discount
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            50,
+            100,
+            98,
+            90,
+            -2,
+            pc(94 * (PD_SCALE as i64), 2 * PD_SCALE, -9)
+        );
+
+        // full deposits, staggered initial discount
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            100,
+            100,
+            98,
+            90,
+            -2,
+            pc(90 * (PD_SCALE as i64), 2 * PD_SCALE, -9)
+        );
+
+        // test precision limits
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            1,
+            1_000_000_000_000_000_000,
+            100,
+            90,
+            -2,
+            pc(100 * (PD_SCALE as i64)-1000, 2 * PD_SCALE, -9),
+        );
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            100_000_000,
+            1_000_000_000_000_000_000,
+            100,
+            90,
+            -2,
+            pc(100 * (PD_SCALE as i64)-1000, 2 * PD_SCALE, -9),
+        );
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            1_000_000_000,
+            1_000_000_000_000_000_000,
+            100,
+            90,
+            -2,
+            pc(100 * (PD_SCALE as i64)-1000, 2 * PD_SCALE, -9),
+        );
+        succeeds(
+            pc(100 * (PD_SCALE as i64), 2 * PD_SCALE, -9),
+            10_000_000_000,
+            1_000_000_000_000_000_000,
+            100,
+            90,
+            -2,
+            pc(100 * (PD_SCALE as i64)-1000, 2 * PD_SCALE, -9),
+        );
+        succeeds(
```

Comment:
i think the tests between 1 and 10M are sort of unnecessary. 1 and 10M are covering the two endpoints for the behavior, and you know that intervening values will give you the same result.